### ISSUE TITLE
Using custom ILoggerProvider and IServiceFactoryProvider breaks logging scopes of WebHost

### ISSUE THREAD

**Issue by muratg** [aspnet/Hosting#1536]
_From @schuettecarsten on Wednesday, 08 August 2018 05:19:33_

Create a sample AspNetCore application. Add a custom logging Framework that implements ILoggerProvider. Observe the behaviour of logging scopes, for each request you get several scopes that contain details about the connection and the request.

Now add a custom DI container framework that implements IServiceFactoryProvider and a observe again. The logging scopes are lost.

You can see that your LoggerProvider implementation is initialized twice. One time immediately after the start of the application, when the WebHost is initialized. After that, the custom DI container is registered and the logger is initialized again. Also, a new IExternalScopeProvider is registered at the LoggerProvider.

The problem seems to be that the WebHost still uses the old, orphaned scope provider.

_Copied from original issue: aspnet/Home#3395_

**Comment by muratg** [comment]
_From @davidfowl on Wednesday, 08 August 2018 05:53:02_

This might be related to these changes https://github.com/aspnet/Hosting/commit/6b095cf533950e7fc93ebc475093bafac70058cb

cc @pakrym

**Comment by davidfowl** [comment]
@schuettecarsten do you have a repro application?

**Comment by schuettecarsten** [comment]
> @schuettecarsten do you have a repro application?

Yes, but it's a bit large because of the 3rd party DI container and logger classes. I can send you a download link directly, please send me a message.

**Comment by davidfowl** [comment]
Nope, I forgot about this one. @schuettecarsten are you still having issues and have you worked around them??
